src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -760,6 +765,16 @@ internal static Type GetDataContractAdapterType(Type type)
                 return type;
             }
 
+            // Maps adapted types back to the original type
+            // Any change to this method should be reflected in GetDataContractAdapterType
+            internal static Type GetDataContractOriginalType(Type type)
+            {
+                if (type == Globals.TypeOfDateTimeOffsetAdapter)
```

Comment:
We have another adapter type, KeyValuePairAdapter. Can you verify whether we need to do the same thing for it?